osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -332,13 +350,20 @@ public TModel Import(TModel item, ArchiveReader archive = null)
                     }
                     catch (Exception e)
                     {
+                        if (!Delete(item))
+                            Files.Dereference(item.Files.Select(f => f.FileInfo).ToArray());
+
                         write.Errors.Add(e);
                         throw;
                     }
                 }
 
                 Logger.Log($"Import of {item} successfully completed!", LoggingTarget.Database);
             }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
             catch (Exception e)
             {
                 Logger.Error(e, $"Import of {item} failed and has been rolled back.", LoggingTarget.Database);
```

Comment:
After this line, `item` is set to `null` which leads to returning a `null` object, so the same comment applies as in my previous comment about `return null;`.

Reply:
This is intentional. This method handled errors and rollback logic itself. All public methods of this class are intended to not throw errors outwards.